src/components/Checkbox/Checkbox.scss

```diff
@@ -1,82 +1,121 @@
+        &::after {
+          background: color('white');
+        }
+      }
     }
-  }
 
-  &:checked,
-  &.Input-indeterminate {
-    // stylelint-disable-next-line selector-max-class
-    ~ .Icon {
-      transform: translate(-50%, -50%) scale(1);
+    &:checked,
+    &.Input-indeterminate {
+      // stylelint-disable-next-line selector-max-class, selector-max-specificity
+      ~ .Icon {
+        transform: translate(-50%, -50%) scale(1);
+      }
+    }
+
+    &:focus,
+    &:active {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop(focused);
+      }
     }
-  }
 
-  &:focus,
-  &:active {
-    + .Backdrop {
-      @include control-backdrop(focused);
+    &:disabled {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop(disabled);
+      }
+      // stylelint-disable-next-line selector-max-class, selector-max-specificity
+      ~ .Icon {
+        @include recolor-icon(color('ink', 'lightest'));
+      }
     }
   }
 
-  &:disabled {
-    + .Backdrop {
-      @include control-backdrop(disabled);
+  &.globalTheming {
+    &:focus {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include focus-ring-animated($show: 1);
+      }
+    }
+    // stylelint-disable-next-line selector-max-specificity
+    &:active:not(:disabled),
+    &:checked,
+    &.Input-indeterminate {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop-dl(active);
+      }
+      // stylelint-disable-next-line selector-max-specificity, selector-max-class
+      ~ .Icon {
+        transform: translate(-50%, -50%) scale(1);
+      }
     }
 
-    ~ .Icon {
-      @include recolor-icon(color('ink', 'lightest'));
+    &:disabled {
+      // stylelint-disable-next-line selector-max-specificity
+      + .Backdrop {
+        @include control-backdrop-dl(disabled);
+      }
+    }
+    // stylelint-disable-next-line selector-max-specificity
+    &:disabled:checked {
+      // stylelint-disable-next-line selector-max-specificity,
+      + .Backdrop {
+        background: var(--p-neutral-border-disabled);
+        // stylelint-disable-next-line selector-max-specificity, selector-max-class, max-nesting-depth, selector-max-combinators
+        &::before {
+          background: var(--p-neutral-border-disabled);
+        }
+      }
     }
   }
 }
 
 .Backdrop {
-  @include control-backdrop;
-  position: absolute;
-  top: 0;
-  bottom: 0;
-  left: 0;
-  right: 0;
-  pointer-events: none;
+  &:not(.globalTheming) {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    pointer-events: none;
+    @include control-backdrop;
+  }
+
+  &.globalTheming {
+    @include control-backdrop-dl;
+    position: relative;
+    display: block;
+    width: 100%;
+    height: 100%;
+    @include focus-ring-animated;
```

Comment:
It doesn't make a lot of sense to me to have a separate mixin for the animated focus ring, since all components that use the focus ring will need it to be animated. I'd like to see one mixin that works for all components which use the focus ring (Button and TextField, for example).